q2cli/core/usage.py

```diff
@@ -0,0 +1,73 @@
+import qiime2.sdk.usage as usage
+
+
+class CLIUsage(usage.Usage):
+    def __init__(self):
+        super().__init__()
+        self._imports = set()
+        self._recorder = []
+        self._init_data_refs = dict()
+
+    def _init_data_(self, ref, factory):
+        self._init_data_refs[ref] = factory
+        # Don't need to compute anything, so just pass along the ref
+        return ref
+
+    def _merge_metadata_(self, ref, records):
+        first_md = records[0].ref
+        remaining_records = ', '.join([r.ref for r in records[1:]])
+        t = '%s = %s.merge(%s)\n' % (ref, first_md, remaining_records)
+        self._recorder.append(t)
+        return ref
+
+    def _get_metadata_column_(self, ref, record, column_name):
```

Comment:
I suspect this will require a little but of additional machinery, behind the scenes. Like metadata merging, getting a column is inlined in the command:

```bash
--m-metadata-file foo.tsv
--m-metadata-column hello
```

So the thing to keep in mind is that in raw usage examples, the `get_metadata_column` invocation will always come before the `action` invocation - so it'll be up to you in this driver to glue those parts together, when the time comes.